hal/targets/hal/TARGET_ONSEMI/TARGET_NCS36510/pwmout_api.c

```
@@ -0,0 +1,210 @@
+	/* Duty cycle shall be 50% and prescaler shall be disabled by default */
+	obj->pwmReg->DUTYCYCLE = 0x80;
+
+	/* Write the PWM output enable register 0x4000B004, to 1 */
+	obj->pwmReg->PWM_ENABLE.WORD = 0x1;
+
+	obj->pwmReg->PRESCALE_DISABLE = 0x1;
+
+}
+
+/** Deinitialize the pwmout object
+ *
+ * @param obj The pwmout object
+ */
+void pwmout_free(pwmout_t *obj)
+{
+	/* Write the PWM output disable register 0x4000B008, to 1 */ 
+	obj->pwmReg->PWM_DISABLE = 0x1;
+}
+
+/** Set the output duty-cycle in range <0.0f, 1.0f>
+ *
+ * Value 0.0f represents 0 percentage, 1.0f represents 100 percent.
+ * @param obj     The pwmout object
+ * @param percent The floating-point percentage number
+ */
+void pwmout_write(pwmout_t *obj, float percent)
+{
+	if (percent == 0.0)
+	{
+		obj->pwmReg->DUTYCYCLE = 0x00;
+	}
+	else if (percent == 1.0)
+	{
+		obj->pwmReg->DUTYCYCLE = 0xFF;
+	}
+	else
+	{
+		/* Write the duty cycle config register 0x4000B000, with the value passed on */
+		/* ((percent * 255) + 1) is the duty cycle. Plus 1 is for accounting for round off errors; like a ceil function */
+		obj->pwmReg->DUTYCYCLE = (uint8_t)((percent * 255) + 1);
+	}
+}
+
+/** Read the current float-point output duty-cycle
+ *
+ * @param obj The pwmout object
+ * @return A floating-point output duty-cycle
+ */
+float pwmout_read(pwmout_t *obj)
+{
+	float retVal = 0.0;
+	float dc = 0.0;
+
+	/* Read out the value of duty cycle register 0x4000B000 and return as a percent */
+	/* Read value / 255 is the percent returned */
+	dc = obj->pwmReg->DUTYCYCLE;
+	retVal = dc/ (float)255;
+
+	return(retVal);
+}
+
+/** Set the PWM period specified in seconds, keeping the duty cycle the same
+ *
+ * Periods smaller than microseconds (the lowest resolution) are set to zero.
+ * @param obj     The pwmout object
+ * @param seconds The floating-point seconds period
+ */
+void pwmout_period(pwmout_t *obj, float seconds)
+{
+	/* Cannot be configured, prescaler is either 256 or 4096 */
```

I wonder if silent error is a good solution (no return value as we cant return, but should we error for this?)